docs: Added contributing guide for developers

paulGeoghegan opened the pull request:
Fixes #486

A contributing guide has now been added with sections to help devs understand how to set up the project and how best to contribute. Places for adding future things like style guidelines and code quality requirements have been included to help make this more expandable in the future.

[//]: # (A few things to note:)
[//]: # (1. Write each changelog as an unordered list in their appropriate sub-heading)
[//]: # (2. The changelogs will be automatically added to `docs/changelogs/latest.md` by the merge workflow)
[//]: # (3. You can write an overview or anything that you don't want to be included as changelog before the 'Changelog' heading)
[//]: # (4. Do not change the heading names and/or the heading levels)
[//]: # (5. Remove the unwanted changelog sections)


## Changelog


### Guides And Docs
- Added contributing guide for developers

paulGeoghegan committed:
docs: Added contributing guide for developers

Fixes #486

A contributing guide has now been added with sections to help devs understand how to set up the project and how best to contribute.
Places for adding future things like style guidelines and code quality requirements have been included to help make this more expandable in the future.

khanshoaib3 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,90 @@
+
+2. **Install dependencies:**
+	 - Ensure you have [.NET 6 SDK](https://dotnet.microsoft.com/download/dotnet/6.0) installed.
+	 - You will need [Ruby](https://www.ruby-lang.org/en/downloads/) for compiling docs (see `docs/compiler_script.rb`).
```
I guess we could also add a point to ensure Project Fluent is installed and the folder's path also match the one specified [here](https://github.com/stardew-access/stardew-access/blob/e2c7dd4088efc7e069daafaca8120c6bd00d6df3/stardew-access/stardew-access.csproj#L30).

paulGeoghegan committed:
docs: Update contributing setup instructions to include Project Fluent

Added steps for installing Project Fluent, which is required for translations and localization. This clarifies the folder structure needed for proper setup.

github-actions[bot] committed:
docs: Added changelogs from #487 to latest.md
